.github/workflows/deploy.yml

```diff
@@ -48,7 +48,23 @@ jobs:
         if: ${{ steps.cache.outputs.cache-hit != 'true' }}
         run: npm ci
 
+      # Cache OG images across builds for performance. This uses a hybrid approach:
+      # 1. GitHub Actions caches the entire public/ directory (directory-level)
+      # 2. Build-time logic in quartz/plugins/emitters/ogImage.tsx checks each file's
+      #    modification time and skips regeneration if cached image is newer than source
+      # 3. Only changed files regenerate their OG images, others reuse cached versions
+      # This provides per-file granularity without managing thousands of cache entries.
+      - name: Cache OG Images
+        uses: actions/cache@v4
+        with:
+          path: public
+          key: og-images-${{ github.sha }}
+          restore-keys: |
+            og-images-
```

Comment:
How does this restore key work? The cache key includes the git hash but this doesn't...

Also (separately) do we need to do anything special so that the cache will be updated with the newly generated images (generated at runtime) or does this already do the right thing to ensure that any files in that directory when the build is complete will be used for the newer cache version?